lib/std/json.zig

```diff
@@ -1376,11 +1376,9 @@ pub const Parser = struct {
     }
 
     fn parseString(p: *Parser, allocator: *Allocator, s: std.meta.TagPayloadType(Token, Token.String), input: []const u8, i: usize) !Value {
-        // TODO: We don't strictly have to copy values which do not contain any escape
-        // characters if flagged with the option.
         const slice = s.slice(input, i);
         switch (s.escapes) {
-            .None => return Value{ .String = try mem.dupe(allocator, u8, slice) },
+            .None => return Value{ .String = if (p.copy_strings) try mem.dupe(allocator, u8, slice) else slice },
```

Comment:
This means that strings with escapes in them are copied, and those without escapes are not. The caller would have no indication of that fact, so how would they free them?

Reply:
If you assume an arena allocator then you don't need such an indication. Does the implementation set up an arena allocator? (Genuine question; I'm typing this on a phone and don't remember )

Reply:
It does use an arena allocator.